### PR TITLE
Support self-hosted runner groups

### PR DESCRIPTION
**Pull request by renjithr23**
relates to issue #1594

### CODE REVIEW

File: github/actions_runner_groups.go
Comment (gmlewis) [inline github/actions_runner_groups.go:1]:
```suggestion
// Copyright 2021 The go-github AUTHORS. All rights reserved.
```

File: github/actions_runner_groups.go
Comment (gmlewis) [inline github/actions_runner_groups.go:25]:
```suggestion
// RunnerGroups represents a collection of self-hosted runner groups configured for an organization.
```

File: github/actions_runner_groups.go
Comment (gmlewis) [inline github/actions_runner_groups.go:31]:
```suggestion
// CreateRunnerGroupRequest represents a request to create a Runner group for an organization.
```

File: github/actions_runner_groups.go
Comment (gmlewis) [inline github/actions_runner_groups.go:36]:
```suggestion
	SelectedRepositoryIDs []int64 `json:"selected_repository_ids,omitempty"`
```

File: github/actions_runner_groups.go
Comment (gmlewis) [inline github/actions_runner_groups.go:38]:
```suggestion
	Runners []int64 `json:"runners,omitempty"`
```

File: github/actions_runner_groups.go
Comment (gmlewis) [inline github/actions_runner_groups.go:41]:
```suggestion
// UpdateRunnerGroupRequest represents a request to update a Runner group for an organization.
```

File: github/actions_runner_groups.go
Comment (gmlewis) [inline github/actions_runner_groups.go:51]:
```suggestion
	SelectedRepositoryIDs []int64 `json:"selected_repository_ids"`
```

File: github/actions_runner_groups.go
Comment (gmlewis) [inline github/actions_runner_groups.go:54]:
```suggestion
// SetRunnerGroupRunnersRequest represents a request to replace the list of
```

File: github/actions_runner_groups.go
Comment (gmlewis) [inline github/actions_runner_groups.go:58]:
```suggestion
	Runners []int64 `json:"runners"`
```

File: github/actions_runner_groups.go
Comment (gmlewis) [inline github/actions_runner_groups.go:61]:
```suggestion
// ListOrganizationRunnerGroups lists all self-hosted runner groups configured in an organization.
```

File: github/actions_runner_groups.go
Comment (gmlewis) [inline github/actions_runner_groups.go:85]:
```suggestion
// GetOrganizationRunnerGroup gets a specific self-hosted runner group for an organization using its RunnerGroup ID.
```

File: github/actions_runner_groups.go
Comment (gmlewis) [inline github/actions_runner_groups.go:104]:
```suggestion
// DeleteOrganizationRunnerGroup deletes a self-hosted runner group from an organization.
```

File: github/actions_runner_groups.go
Comment (gmlewis) [inline github/actions_runner_groups.go:118]:
```suggestion
// CreateOrganizationRunnerGroup creates a new self-hosted runner group for an organization.
```

File: github/actions_runner_groups.go
Comment (gmlewis) [inline github/actions_runner_groups.go:121]:
```suggestion
func (s *ActionsService) CreateOrganizationRunnerGroup(ctx context.Context, org string, createReq CreateRunnerGroupRequest) (*RunnerGroup, *Response, error) {
```

File: github/actions_runner_groups.go
Comment (gmlewis) [inline github/actions_runner_groups.go:140]:
```suggestion
func (s *ActionsService) UpdateOrganizationRunnerGroup(ctx context.Context, org string, groupID int64, updateReq UpdateRunnerGroupRequest) (*RunnerGroup, *Response, error) {
```

File: github/actions_runner_groups.go
Comment (gmlewis) [inline github/actions_runner_groups.go:156]:
```suggestion
// ListRepositoryAccessRunnerGroup lists the repositories with access to a self-hosted runner group configured in an organization.
```

File: github/actions_runner_groups.go
Comment (gmlewis) [inline github/actions_runner_groups.go:175]:
```suggestion
// SetRepositoryAccessRunnerGroup replaces the list of repositories that have access to a self-hosted runner group configured in an organization
```

File: github/actions_runner_groups.go
Comment (gmlewis) [inline github/actions_runner_groups.go:179]:
```suggestion
func (s *ActionsService) SetRepositoryAccessRunnerGroup(ctx context.Context, org string, groupID int64, ids SetRepoAccessRunnerGroupRequest) (*Response, error) {
```

File: github/actions_runner_groups.go
Comment (gmlewis) [inline github/actions_runner_groups.go:190]:
```suggestion
// AddRepositoryAccessRunnerGroup adds a repository to the list of selected repositories that can access a self-hosted runner group.
```

File: github/actions_runner_groups.go
Comment (gmlewis) [inline github/actions_runner_groups.go:191]:
```suggestion
// The runner group must have visibility set to 'selected'.
```

File: github/actions_runner_groups.go
Comment (gmlewis) [inline github/actions_runner_groups.go:205]:
```suggestion
// RemoveRepositoryAccessRunnerGroup removes a repository from the list of selected repositories that can access a self-hosted runner group.
```

File: github/actions_runner_groups.go
Comment (gmlewis) [inline github/actions_runner_groups.go:206]:
```suggestion
// The runner group must have visibility set to 'selected'.
```

File: github/actions_runner_groups.go
Comment (gmlewis) [inline github/actions_runner_groups.go:220]:
```suggestion
// ListRunerGroupRunners lists self-hosted runners that are in a specific organization group.
```

File: github/actions_runner_groups.go
Comment (gmlewis) [inline github/actions_runner_groups.go:244]:
```suggestion
// SetRunerGroupRunners replaces the list of self-hosted runners that are part of an organization runner group
```

File: github/actions_runner_groups.go
Comment (gmlewis) [inline github/actions_runner_groups.go:248]:
```suggestion
func (s *ActionsService) SetRunerGroupRunners(ctx context.Context, org string, groupID int64, ids SetRunnerGroupRunnersRequest) (*Response, error) {
```

File: github/actions_runner_groups.go
Comment (gmlewis) [inline github/actions_runner_groups.go:259]:
```suggestion
// AddRunerGroupRunners adds a self-hosted runner to a runner group configured in an organization.
```

File: github/actions_runner_groups.go
Comment (gmlewis) [inline github/actions_runner_groups.go:273]:
```suggestion
// RemoveRunerGroupRunners removes a self-hosted runner from a group configured in an organization.
```

File: github/actions_runner_groups_test.go
Comment (gmlewis) [inline github/actions_runner_groups_test.go:1]:
```suggestion
// Copyright 2021 The go-github AUTHORS. All rights reserved.
```

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @renjithr23 !  This is off to a great start.
Please feel free to push as many more commits to this PR as you wish (and please do not use force push) because we will perform a single "Squash and Merge" at the very end when we are ready to merge.

**Comment by renjithr23** [comment]
@gmlewis Thanks for the extensive code review. I made all the changes you've suggested.

File: github/actions_runner_groups.go
Comment (gmlewis) [inline github/actions_runner_groups.go:276]:
```suggestion
// GitHub API docs: https://docs.github.com/en/rest/reference/actions#remove-a-self-hosted-runner-from-a-group-for-an-organization
```

Comment (renjithr23) [reply github/actions_runner_groups.go]: I guess I need to be a little more careful. Thanks again!! @gmlewis . I made the change

Comment (gmlewis) [reply github/actions_runner_groups.go]: > I guess I need to be a little more careful. Thanks again!! @gmlewis . I made the change

No worries!  That's what code reviews are for.  :joy:

**Review by gmlewis — APPROVED** [review]
Thank you, @renjithr23 !
LGTM.

Awaiting second LGTM before merging.

**Review by wesleimp — APPROVED** [review]
LGTM 👌🏼

**Comment by gmlewis** [comment]
Thank you, @wesleimp !
Merging.

**Pull request merged**
